### ISSUE TITLE
DOWNLOAD_MIRROR behavior has changed

### ISSUE THREAD

**Issue by stevenhair** [nodkz/mongodb-memory-server#512]
## Versions

- NodeJS: 14.17.3
- mongodb-memory-server-*: 7.3.1
- mongodb(the binary version): 0.0.0
- system: 

package: mongo-memory-server-core

## What is the Problem?

25c865c0a71e7d682eccb6f0e2895a11f9f092ef changed the behavior of the `DOWNLOAD_MIRROR` option - it is no longer possible to specify a path prefix.

## Code Example

```bash
export MONGOMS_DOWNLOAD_MIRROR=https://artifactory.mydomain.com/artifactory/mongo-fastdl
npm install
npm test
```

The above results in:

```
     Error: Status Code isnt 200!
      at ClientRequest.<anonymous> (node_modules/mongodb-memory-server-core/src/util/MongoBinaryDownload.ts:391:20)
      at HTTPParser.parserOnIncomingClient (_http_client.js:641:27)
      at HTTPParser.parserOnHeadersComplete (_http_common.js:126:17)
      at TLSSocket.socketOnData (_http_client.js:509:22)
      at addChunk (internal/streams/readable.js:309:12)
      at readableAddChunk (internal/streams/readable.js:284:9)
      at TLSSocket.Readable.push (internal/streams/readable.js:223:10)
      at TLSWrap.onStreamRead (internal/stream_base_commons.js:188:23)
      at TLSWrap.callbackTrampoline (internal/async_hooks.js:131:14)
```

## Do you know *why* it happens?

The URL that is being hit is https://artifactory.mydomain.com/linux/mongodb-linux-x86_64-rhel70-4.0.25.tgz (it should be https://artifactory.mydomain.com/artifactory/mongo-fastdl/linux/mongodb-linux-x86_64-rhel70-4.0.25.tgz)

This replaces the path on the download mirror:

https://github.com/nodkz/mongodb-memory-server/blob/25c865c0a71e7d682eccb6f0e2895a11f9f092ef/packages/mongodb-memory-server-core/src/util/MongoBinaryDownloadUrl.ts#L55-L56

I think that the code needs to be a bit smarter about how it adds the path suffix - maybe something like this:

```typescript
 const url = new URL(`${mirror.replace(/\/$/, '')}/${this.platform}/${archive}`); 
 return url.toString()
```

**Comment by hasezoey** [comment]
from what i can tell (earlier documentation and such (pre 7)), it was never intended or thought of that the config option `DOWNLOAD_MIRROR` is more than the protocol + address (no extra path)

will add an fix later

**Comment by stevenhair** [comment]
The documentation doesn't mention it, but the original feature request in #91 is the same as my use case. Unless you have a reverse proxy sitting in front of Artifactory, everything will have a path prefix.

Even if that wasn't the intention, it would still be a nice feature to have. I could specify the full download URL rather than just the mirror, but that has other challenges when it comes to supporting multiple platforms.

**Comment by hasezoey** [comment]
this issue should be fixed with ae7ddab79fcbb637cdcd2c193554e9ba2907cf9e for runtime and 6a1636d4e319b27588a62cf66dfb8226b97d2022 for documentation

also there is #513

**Comment by stevenhair** [comment]
Thanks for the quick response!
